src/NumericalAlgorithms/LinearSolver/Gmres.cpp

```diff
@@ -0,0 +1,95 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "NumericalAlgorithms/LinearSolver/Gmres.hpp"
+
+#include <blaze/math/Column.h>
+#include <blaze/math/Submatrix.h>
+#include <blaze/math/Subvector.h>
+#include <blaze/math/lapack/trsv.h>
+
+#include "DataStructures/DenseMatrix.hpp"
+#include "DataStructures/DenseVector.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace LinearSolver {
+namespace gmres_detail {
+
+namespace {
+
+void update_givens_rotation(const gsl::not_null<double*> givens_cosine,
+                            const gsl::not_null<double*> givens_sine,
+                            const double rho, const double sigma) noexcept {
+  if (UNLIKELY(rho == 0.)) {
+    *givens_cosine = 0.;
+    *givens_sine = 1.;
+  } else {
+    const double tmp = sqrt(square(rho) + square(sigma));
+    *givens_cosine = abs(rho) / tmp;
+    *givens_sine = *givens_cosine * sigma / rho;
+  }
+}
+
+template <typename Arg>
+void apply_and_update_givens_rotation(
+    Arg&& argument,
```

Comment:
I couldn't get that working with the `blaze::Column` that is a view into the matrix.. happy to fix this if you have an idea?

Reply:
Works for me with `Arg`.  What problem are you seeing?

Reply:
`Arg` works. was confused because the argument should be a reference, but that's somehow built into the `blaze::View<blaze::Column>` type. I wanted to take the `argument` as a `blaze::DenseVector<T, TF>&` or some other trickery that allows taking any blaze vector reference, including a view into a matrix column, but couldn't get that working. Since it doesn't even work for blaze-native functions like `blaze::trsv` it may not be possible. Anyway, since this is only an implementation detail in an anonymous namespace it doesn't really matter.